src/highdicom/map/sop.py

```diff
@@ -0,0 +1,789 @@
+                )
+            return (
+                _PixelDataType.USHORT,
+                self._pixel_data_type_map[_PixelDataType.USHORT],
+            )
+        elif pixel_array.dtype.kind == "i":
+            if pixel_array.dtype not in (np.int8, np.int16):
+                raise ValueError(
+                    'Unsupported signed integer type for pixel data: '
+                    '8-bit or 16-bit signed integer types are supported.'
+                )
+            return (
+                _PixelDataType.SHORT,
+                self._pixel_data_type_map[_PixelDataType.SHORT],
+            )
+        raise ValueError(
+            'Unsupported data type for pixel data.'
+            'Supported are 8-bit or 16-bit signed and unsigned integer types '
+            'as well as 32-bit (single-precision) or 64-bit (double-precision) '
+            'floating-point types.'
+        )
+
+    def _encode_pixels(self, plane: np.ndarray) -> bytes:
+        """Encodes a given pixel array as a bytes object
+
+        Parameters
+        ----------
+        plane : np.ndarray
+            The numpy array to encode
+
+        Returns
+        -------
+        bytes
+            `plane` encoded as a `bytes` object
+
+        Raises
+        ------
+        ValueError
+            If the SOP instance uses an encapsulated transfer syntax and
+            `plane` is not exactly 2 dimensional.
+        """
+        if self.file_meta.TransferSyntaxUID.is_encapsulated:
+            # Check that only a single plane was passed
+            if plane.ndim != 2:
+                raise ValueError(
+                    'Only single frame can be encoded at at time '
+                    'in case of encapsulated format encoding.'
+                )
+            return encode_frame(
+                plane.astype(np.uint16),
+                transfer_syntax_uid=self.file_meta.TransferSyntaxUID,
+                bits_allocated=self.BitsAllocated,
+                bits_stored=self.BitsStored,
+                photometric_interpretation=self.PhotometricInterpretation,
+                pixel_representation=self.PixelRepresentation,
+            )
+        else:
+            if plane.dtype == np.uint8:
+                return plane.astype(np.uint16).flatten().tobytes()
+            elif plane.dtype.kind == 'i':
+                plane = plane.astype(np.int16) + 2 ** 16 / 2
+                return plane.astype(np.uint16).flatten().tobytes()
+            else:
+                return plane.flatten().tobytes()
+    
+    @property
+    def pixel_array(self) -> np.ndarray:
+        """This is a workaround for a bug in Pydicom.
+
+        In PyDicom you are required to set PixelRepresentation in order to
+        use the numpy pixel_data handler, however this goes against the DICOM
+        standard. To get around this issue we set the Pixel Representation
+        attribute if it is not set, then get the array, then remove the
+        attribute.
```

Comment:
I appreciate the detailed comment, but as written this docstring for this public property will go into the sphinx generated user documentation. The docstring should be written for users and hide this sort of detail, and comments like this should probably be code comments at the top of the method.

Reply:
Resolving this conversation, since it's outdated.